test/Tests/src/Data/Time/Date_Part_Spec.enso

```diff
@@ -0,0 +1,99 @@
+from Standard.Base import all
+
+import Standard.Test
+
+spec name create_new_date =
+    Test.group (name + "date part tests")  <|
+        Test.specify "should return if a leap year" <|
+            create_new_date 2022 8 25 . is_leap_year . should_equal False
+            create_new_date 1999 12 31 . is_leap_year . should_equal False
+            create_new_date 1996 6 19 . is_leap_year . should_equal True
+            create_new_date 2000 1 1 . is_leap_year . should_equal True
+            create_new_date 1900 1 1 . is_leap_year . should_equal False
+
+        Test.specify "should return the number of days in the year" <|
+            create_new_date 2022 8 25 . length_of_year . should_equal 365
+            create_new_date 1999 12 31 . length_of_year . should_equal 365
+            create_new_date 1996 6 19 . length_of_year . should_equal 366
+            create_new_date 2000 1 1 . length_of_year . should_equal 366
+            create_new_date 1900 1 1 . length_of_year . should_equal 365
+
+        Test.specify "should return the century" <|
+            create_new_date 2022 8 25 . century . should_equal 21
+            create_new_date 1999 12 31 . century . should_equal 20
+            create_new_date 2000 1 1 . century . should_equal 20
+            create_new_date 2001 1 1 . century . should_equal 21
```

Comment:
This cries for a test for years `1 AD` and `1 BC` and some other BC years. I'm not sure how well `LocalDate` support for these works - I was just able to construct `LocalDate.of(0, 10, 12)` whereas year 0 does not exist in the Gregorian calendar.

Still, if we don't want to support years prior to some period, according to our philosophy of preferring errors over data corruption we should throw an error that the operation is not supported. Otherwise some users will use the tool for ancient years and instead of getting an explicit errors they will get wrong results - which is really really bad.

Reply:
Limited to 1 AD onwards now. We should at some point pick a uniform minimum allowed date.

Excel struggles before 1st March 1900.
Gregorian calendar introduced on 15th October 1582 might make sense too.